### PR TITLE
Removing Unknown Error

### LINKED ISSUE #6422: Find source of "Unknown error" errors

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #6422.

### CODE REVIEW

File: src/service/resource.js [hunk 1]
```diff
@@ -150,6 +150,9 @@ export class Resource {
     /** @private {number} */
     this.layoutCount_ = 0;
 
+    /** @private {*|null} */
+    this.lastLayoutError_ = null;
```
Comment (erwinmombay) [inline src/service/resource.js]: shouldn't be null already be covered by `*`?

File: src/error.js [hunk 2]
```diff
@@ -54,18 +54,35 @@ let globalExponentialBackoff = function(work) {
...
+  } else {
+    error = new Error('Unknown error');
   }
-  if (!error) {
-    error = new Error('no error supplied');
+  // Report if error is not an expected type.
+  if (!isValidError && getMode().localDev) {
+    setTimeout(function() {
+      const rethrow = new Error('Error reported incorrectly: ' + error);
+      rethrow.reported = true;
```
Comment (jridgewell) [inline src/error.js]: Will this stop the rethrow from being logged in the console?

File: src/error.js [hunk 3]
```diff
@@ -54,18 +54,35 @@ let globalExponentialBackoff = function(work) {
...
  * @param {!Element=} opt_associatedElement
+ * @return {!Error}
  */
 export function reportError(error, opt_associatedElement) {
-  if (!self.console) {
-    return;
+  // Convert error to the expected type.
+  let isValidError;
+  if (error) {
+    if (typeof error == 'string') {
```
Comment (cramforce) [inline src/error.js]: Maybe check for `object` and `!null` instead? Errors could also be numbers and other crap.

File: src/error.js (hunk 2 above)
Comment (dvoytenko) [inline src/error.js]: Hmm. I do see it in console right now. But I changed it to do `_reported_` - that one is only ignored for sending to server.

File: src/service/resource.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/resource.js]: Correct. Fixed.

File: src/error.js (hunk 3 above)
Comment (dvoytenko) [inline src/error.js]: Done

**Pull request merged**